File: app/src/main/java/org/zeroxlab/imgorg/lib/MockOperation.java
package org.zeroxlab.imgorg.lib;

import android.content.Context;
import android.util.Log;

import java.text.SimpleDateFormat;
import java.util.Date;

public class MockOperation implements Operation {

    private final static SimpleDateFormat sSDF = new SimpleDateFormat("yyyy-MM");

    private Media mMedia;
    private String mDest;
    private boolean mFinished = false;

    MockOperation(Media src, String dest) {
        mMedia = src;
        Date d = src.date == null ? new Date() : src.date;
        String dstDate = sSDF.format(d);
        mDest = dest + "/" + dstDate;
    }

    @Override
    public void consume(Context ctx) {
        mFinished = true;
        Log.d("Mock", "consumed operation");
    }

    @Override
    public boolean isFinished() {
        return mFinished;
    }

    @Override
    public String getSource() {
        return mMedia.data;
    }

    @Override
    public String getDestination() {
        return mDest;
    }
}

File: app/src/main/java/cc/jchu/imgorg/lib/MockOperation.kt
package cc.jchu.imgorg.lib

import android.content.Context
import android.util.Log
import java.text.SimpleDateFormat
import java.util.Date

class MockOperation internal constructor(private val mMedia: Media, dest: String) : Operation {
    override val destination: String
    override var isFinished = false
        private set

    override fun consume(ctx: Context?) {
        isFinished = true
        Log.d("Mock", "consumed operation")
    }

    override val source: String?
        get() = mMedia.data

    companion object {
        private val sSDF = SimpleDateFormat("yyyy-MM")
    }

    init {
        val d = mMedia.date ?: Date()
        val dstDate = sSDF.format(d)
        destination = "$dest/$dstDate"
    }
}
